dali/pipeline/workspace/workspace.h

```diff
@@ -526,20 +527,95 @@ class WorkspaceBase : public ArgumentWorkspace {
   }
 
   /**
- * @brief Returns the index of the sample that this workspace stores
- * in the input/output batch.
- */
+   * @brief Returns the index of the sample that this workspace stores
+   * in the input/output batch.
+   */
   DLL_PUBLIC virtual inline int data_idx() const {
     return 0;
   }
 
   /**
- * @brief Returns the index of the thread that will process this data.
- */
+   * @brief Returns the index of the thread that will process this data.
+   */
   DLL_PUBLIC virtual inline int thread_idx() const {
     return 0;
   }
 
+
+  /// @{
+  /**
+   * Sets the operator ID that this Workspace in associated with.
+   */
+  void SetOperatorId(std::string operator_id) {
```

Comment:
One more thing, why do we call it ID if it's instance_name? Should we keep this naming?

Reply:
I'd say, that `id` and `instance_name` can be two different concepts and it's just a coincidence that they have the same value for now. ID would uniquely identify every operator, while I don't know if it's a good idea to guarantee that for `instance_name`. Furthermore, `instance_name` is what Executor knows, while ID is something that Workspace is aware about. As I said, from my perspective these are two separate things.

If you really insist I'll change to `SetOperatorInstanceName`, but I'd prefer to keep the ID. It's shorter...

Reply:
What I mean here, we set it to instance name. Executor also has operator id, but it's just some integer identifying the operator node in the graph.